### ISSUE TITLE
Physical Entity link to Collecting Event

### ISSUE THREAD

**Issue by cgendreau** [DINA-Web/dina-model-concepts#46]
Current component description says:

> Has links to:
> Collection Event(s): a physical entity comes from 1 or multiple collecting event

We need to add more details or change it to only 1. I'm not sure was the justification since a physical item can only, and only, come from 1 collecting event (if you mix 2 things I would say it's a preparation of 2 separate physical entities). The only case that would not fit that could be uncertainty (the entity is from one of these collecting event but I can't say which one). If it's the case, a collecting event expressing the uncertainty should be created (see [Collecting Event](https://github.com/DINA-Web/dina-model-concepts/blob/master/component_descriptions/collecting_event.md))

**Comment by jmacklin** [comment]
I agree based on our discussion. There will always be some outlying use case that will prove this wrong but as you say we can find different ways to indicate "mixes". I do think this is a question we should bring to the focus group to insure that we are not missing something (especially related to environmental samples). I agree with the logic of mixes being a preparation of 2 separate physical entities. Collecting events can be quite variably implemented by different organism types but typically this is constrained by time or space. For example, to a botanist every plant they sample is a separate event even if they are right beside each other. In Zoology it is common to define an area (and sometimes a period of time) that is surveyed and sampled that is defined as one event. In marine sampling, a ship may trawl a 100km transect and just catch whatever they can, so again this is one event. Not exactly sure how Paleo works but Falko can add to this.

**Comment by falkogloeckler** [comment]
Yes, mixes were the issue, we tried to address here. But thinking with a fresh mind we can keep the flexibility even with a one-to-one relation between a physical entity and collecting event if we allow for two things:
* A collecting event could have several sub-events and a physical entity  (aka sample) could have sub-entities (could be easily realized by self-referencing tables without additional complexity in the data model). So nested sampling can be accommodated if needed and each physical entity would still have only one related collecting event. This would cover nested sampling methods with separate the time, date and location entries and without focing into one event.
* If physical entities (regardless of the sampling method, nested or not) are mixed afterwards for whatever scientific reason, this is actually a treatment or preparation that needs to be documented separately. So, latest at the time of preparation (mixing the entities) a retrospective parent collecting event needs to be created (automatically by the system, without any user interaction). So the aggregated entity could be assigned to the parent collecting event.

Is less complicated as it actually reads   ;)

**Comment by cboelling** [comment]
Conceptually, a *Physical Entity* was collected in exactly one *Collecting Event* and I think [James' post](https://github.com/DINA-Web/dina-model-concepts/issues/46#issuecomment-719513377) very nicely illustrates that in different contexts a particular *Collecting Event* can be conceptualized in substantially different ways.

Following our recent discussion I see two problem cases:

1. **One item, composed of two or more items from different *Collecting Events***
The item *is in actuality* tied to more than one *Collecting Event* by virtue of its component items. The relation *is actually* 1:N. It is possible to implement this as a 1:1 relation by creating an additional *Collecting Event* encompassing the individual collecting events. In some cases this new *Collecting Event* may feel natural (e.g., day 1 and day 2 of the week long expedition) in other cases they will feel artificial (e.g., day 1 and day 5 of the expedition). An alternative is to link compound items only via their component parts to *Collecting Events* but this, I think, runs into problems whenever the parts that were actually collected in different events cannot or will not be identified as individual items.

1. **One item for which the origin is uncertain and all eligible *Collecting Events* should be linked to the item**
The item *is in actuality* only tied to a single *Collecting Event* but we don't know which. Conceptually, this requires a different relation between item and *Collecting Event* to represent this state of affairs e.g., *has-potentially-related-collecting-event*. This, again, conceptually is a 1:N relation which can be condensed to a 1:1 relation by creating surrogate *Collecting Events* (which to me appears rather artifical in this case)

From a conceptual design perspective I would choose to implement this, consequently, as a 1:N relation. The alternative, realizing a 1:1 relation through surrogate *Collecting Events*, is a possibility but to me feels artificial and more error-prone.

**Comment by gnewton** [comment]
@cgendreau 
>_One item for which the origin is uncertain and all possible Collecting Events should be linked to the item_
With some items associated with no Collecting Event because we have no idea how we got this item?

**Comment by cboelling** [comment]
> 
> 
> @cgendreau
> 
> > _One item for which the origin is uncertain and all possible Collecting Events should be linked to the item_
> > With some items associated with no Collecting Event because we have no idea how we got this item?

Yes that is a possibility too. I changed [in the description above](https://github.com/DINA-Web/dina-model-concepts/issues/46#issuecomment-729642661) the term "possible" to "eligible" to make the case clearer.
In these cases, by the way, the unknown collecting event (which must exist, we just don't know anything about it) can still be identified by an instance of *Collecting Event* that we just cannot say anything about. This instance can then be used to represent the connection of multiple items to the same, yet unknown, *Collecting Event*.

**Comment by cgendreau** [comment]
> 1. One item, composed of two or more items from different Collecting Events

What if there is simply no Collecting Events on the composed object since we are putting 2 (or more) different things together. Isn't an issue introduced by merging Collection Item and Physical Entities? I'm not convinced that the fact that 2 different things are "prepared" together should go up to the Collecting Event.

**Comment by cboelling** [comment]
> 
> 
> > 1. One item, composed of two or more items from different Collecting Events
> 
> What if there is simply no Collecting Events on the composed object since we are putting 2 (or more) different things together. Isn't an issue introduced by merging Collection Item and Physical Entities? I'm not convinced that the fact that 2 different things are "prepared" together should go up to the Collecting Event.

I guess the correct way to go about this is acknowledging that the relation of a compound item that is potentially a mix of very different items to a *Collecting Event* is materially different from the relation that the component items have to "their" *Collecting Events*. The latter were really collected in these events, the former is related to these events merely by its component parts. It was never actually collected.

Consequently, compound items should be linked to *Collecting Events* only via their component parts or, if we are unable or unilling to represent these component parts, a different relation to link the compound item and the corresponding *Collecting Events* should be used.

**Comment by falkogloeckler** [comment]
> One item, composed of two or more items from different Collecting Events

I'd say we won't have issues if both, the physical entities and collecting events concepts allow for self-referencing parent-child relations. 
Thus, if two physical entities with different collecting events are going to be merged a new entity needs to be created (this needs to be done anyways because of the physical change). The resulting physical entity (parent records) would hold the references to the original entities (child records) and would not necessarily hold a "real" collecting event but maybe a surrogate entry, because the parent was not collected as such (as @cboelling already mentioned). 
Instead, the original collecting events would still be referenced at the level of the child records. 

BTW: If both were collected in the same collecting event and then merged later we should follow the same approach in order to have a clean history of records.

**Comment by jmacklin** [comment]
I agree with Falko. We just need to make sure that the model can address the relations required.

**Comment by cboelling** [comment]
This basically reiterates the point I made earlier so I think this is a viable way of modelling it.

**Comment by cgendreau** [comment]
So if I followed correctly we should update the following:

> Collection Event(s): a physical entity comes from 1 or multiple collecting event

**Comment by falkogloeckler** [comment]
No. What I mean is:  A physical entity would only have one collecting event. But for the potential case of merging entities to one physical entity would be solved by parent-child relations between physical entities (each having a collecting event). Thus, no multiple collecting events are necessary anymore.

**Comment by cgendreau** [comment]
like that? https://github.com/DINA-Web/dina-model-concepts/pull/56

**Comment by falkogloeckler** [comment]
Yes! I merged #56
